docs/ipv6-design-doc.md

```diff
@@ -0,0 +1,164 @@
+
+  * The mechanism for determining the current IPv4 address (the `use`
+    option) must be independently configurable from the mechanism used
+    to determine the current IPv6 address.
+  * The user must be able to disable IPv4 address updates without
+    affecting IPv6 updates.
+  * The user must be able to disable IPv6 address updates without
+    affecting IPv4 updates.
+  * If HTTP polling is used for both IPv4 and IPv6 address discovery,
+    the URL used to determine the IPv4 address (the `web` option) must
+    be independently configurable from the URL used to determine the
+    IPv6 address.
+  * The use of IPv4 or IPv6 to update a record must be independent of
+    the type of record being updated (IPv4 or IPv6).
+  * The callback for the update protocol must be given both addresses,
+    even if only one of the two addresses has changed.
+  * The callback for the update protocol must be told which addresses
+    have changed.
+
+### Nice-to-Haves
+
+  * The user should be able to force the update protocol to use IPv4
+    or IPv6.
+  * Unless configured otherwise, ddclient should first attempt to
+    update via IPv6 and fall back to IPv4 if the IPv6 connection
+    fails. This behavior can be added later; for now it is acceptable
+    to keep the current behavior (use IPv6 without IPv4 fallback if
+    there is a AAAA record, use IPv4 if there is no AAAA record).
+  * Full backwards compatibility with existing config files and
+    flags. The trade-off between migration burden and long-term
+    usability should be carefully considered.
+
+## Proposal
+
+### Configuration changes
+
+  * Add new `usev4` and `usev6` settings that are like the current
+    `use` setting except they only apply to IPv4 and IPv6,
+    respectively.
+      * `usev4` can be set to one of the following values: `ipv4`,
+        `webv4`, `fwv4`, `ifv4`, `cmdv4`, `ciscov4`, `cisco-asav4`
+      * `usev6` can be set to one of the following values: `ipv6`,
+        `webv6`, `fwv6`, `ifv6`, `cmdv6`, `ciscov6`, `cisco-asav6`
+  * Add new `ipv4`, `ipv6`, `webv4`, `webv6`, `ifv4`, `ifv6`,
+    etc. settings that behave like their versionless counterparts
+    except they only apply to IPv4 or IPv6. Deprecate the versionless
+    counterparts, and change their behavior to influence the default
+    value of the versioned options. (Example: Suppose `ifv4=ip`. If
+    `ifv4` is not set then `if` is used.) Special notes:
+      * The value of `ip` will only serve as the default for `ipv4`
+        (or `ipv6`) if it contains an IPv4 (or IPv6) address.
+      * There is currently an `ipv6` boolean setting. To preserve
+        backward compatibility with existing configs, `ipv6` set to a
+        boolean value is ignored (other than a warning).
+  * Deprecate the `use` setting. Change `use=foo` to control the
+    default for `usev4` and `usev6`:
+      * If neither `usev4` nor `usev6` is set, then `use=foo` is
+        equivalent to `usev4=foov4,usev6=foov6`.
```

Comment:
I disagree.  I think the depreciated settings should be equivalent to the new v4 settings.  So `use=foo` is equivalent to `usev4=foov4, usev6=no`. If we don't do this then everyone is going to suddenly be trying to update a IPv6 address at their DDNS server when they probably didn't expect it -- potentially causing errors.